src/instructlab/model/backends/vllm.py

```diff
@@ -120,32 +164,56 @@ def run_vllm(
         background (bool):            Whether the stdout and stderr vLLM should be sent to /dev/null (True)
                                       or stay in the foreground(False).
     Returns:
-        vllm_process (subprocess.Popen): process of the vllm server
+        tuple: A tuple containing two values:
+            vllm_process (subprocess.Popen): process of the vllm server
+            tmp_files: a list of temporary files necessary to launch the process
+
     """
     vllm_process = None
     vllm_cmd = [sys.executable, "-m", "vllm.entrypoints.openai.api_server"]
+    tmp_files = []
 
-    if "--host" not in vllm_args:
+    if not contains_argument("--host", vllm_args):
         vllm_cmd.extend(["--host", host])
 
-    if "--port" not in vllm_args:
+    if not contains_argument("--port", vllm_args):
         vllm_cmd.extend(["--port", str(port)])
 
-    if "--model" not in vllm_args:
+    if not contains_argument("--model", vllm_args):
         vllm_cmd.extend(["--model", os.fspath(model_path)])
 
+    if not contains_argument("--chat-template", vllm_args):
+        if chat_template == CHAT_TEMPLATE_AUTO:
+            # For auto templates, the build-in in-memory template
+            # needs to be written to a temporary file so that vLLM
+            # can read it
+            template = format_template(model_family, model_path)
+            file = create_tmpfile(template)
+            tmp_files.append(file)
```

Comment:
I still don't understand why we need a file where vLLM supports to be given a string directly. @n1hility did I miss the reason somewhere?

Reply:
@leseb yeah the big reason is just that files as arguments run up against problems: parsing/escaping issues, ARG_MAX limits, and cut and paste problems (e..g message is printed unescaped, which won't work with paste, user fixes but gets it wrong, etc). 

In the case where the args are passed to exec(), escaping usually isn't a problem, however, it does rely on the receiver of the args to handle them correctly. For example if the called code use an argument parser which optionally splits on = or some other form of interpolation, that can easily break with a template string in a param. Or if there is some sort of intermediary (for example, they construct a command line which is passed as a shell string). Some OSes also requote (e.g. windows) which can lead to oddities.  

Some projects that do this sort of thing use base64 encoding to avoid the problems (with the exception of ARG_MAX limits)